### ISSUE TITLE
KubeEdge nodes get role=master

### ISSUE THREAD

**Issue by siredmar** [kubeedge/kubeedge#2977]
**What happened**:
Adding nodes to the cluster results in the following roles:
```
$ kubectl get nodes  
NAME                                STATUS   ROLES               AGE   VERSION
aks-agentpool-23303655-vmss000000   Ready    agent               41h   v1.19.11
node0                               Ready    agent,edge,master   60m   v1.19.3-kubeedge-v1.7.1
node1                               Ready    agent,edge,master   27m   v1.19.3-kubeedge-v1.7.1
node2                               Ready    agent,edge,master   27m   v1.19.3-kubeedge-v1.7.1
node3                               Ready    agent,edge,master   27m   v1.19.3-kubeedge-v1.7.1
```

All my nodes have the role=master meaning that on the nodes will run some master related stuff.
Looking at the nodes labels and roles presents this:
```
$ kubectl describe node node0                
Name:               node0
Roles:              agent,edge,master
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    edgefarm.tenant=tenant1
                    kubernetes.azure.com/role=master
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=node0
                    kubernetes.io/os=linux
                    kubernetes.io/role=master
                    node-role.kubernetes.io/agent=
                    node-role.kubernetes.io/edge=
                    node-role.kubernetes.io/master=
```

**What you expected to happen**:
I'd expect that my nodes won't get the master role.

**Environment**:
I have two clusters running in Azure AKS: `main` and `seed`. The `main` cluster runs Kubermatic Kubernetes Platform (KKP). The `seed` cluster is managed by KKP allowing me to add custom nodes to an the `seed` cluster. 
The `seed` cluster runs `cloudcore`. Adding KubeEdge nodes and deploying workload on them basically is working pretty well. 

- KubeEdge version: v1.7.1

**Comment by sirjasiu** [comment]
Hi @siredmar - did you figure out how to deal with this master role? Can you please help us, or share your experience on running kubeedge with aks?

**Comment by siredmar** [comment]
TL;DR kubeedge doesn't run on AKS, like any other (at least the great majority) of hosted k8s providers. They manage the control-plane and forbid adding custom nodes. Why? Because this is the business model. If you need a different amount of compute power, scale your nodes up or down. The bad thing is, that kubeedge nodes behave just like cluster nodes. So this is impossible to use. 

You can use Azure, Google, AWS or any (again at least the great majority) other cloud provider to create VMs and host your kubernetes cluster on your own. For cluster management you can use one of the several projects out there, e.g. Rancher, Gardener, ... or install it manually using Talos, RKE1, RKE2, ...

So in the end **you have to manage your cluster by yourself and cannot use a managed k8s cluster from any cloud provider.**

**Comment by sirjasiu** [comment]
Thanks 😁

**Comment by ageoffron** [comment]
@siredmar This is not my experience. Kubeedge has been working fine with AKS.

@sirjasiu Create your edge node with the necessary labels before you join the node.  This will prevent inherently AKS controlplane from pushing pods.

-```
--
apiVersion: v1
kind: Node
metadata:
  name: "your node name"
  labels:
    kubernetes.azure.com/role: "agent"
    kubernetes.io/role: "agent"
    node-role.kubernetes.io/agent: ""
    node-role.kubernetes.io/edge: ""
    edge.project.com/location: "remote"
spec:
  taints:
  - effect: NoSchedule
    key: "edge.project.com/remote"

```

**Comment by mrchypark** [comment]
Do you have an example of using CloudCore in AKS (Azure Kubernetes Service)?

@ageoffron
